item.rb

```diff
@@ -0,0 +1,32 @@
+require 'date'
+require_relative 'label'
+
+class Item
+  attr_accessor :genre, :author, :source, :label, :publish_date
+  attr_reader :id, :archived
+
+  def initialize(genre, author, source, label, publish_date)
+    @id = Random.rand(1..1000)
+    @genre = genre
+    @author = author
+    @source = source
+    @label = label
+    @publish_date = Date.parse(publish_date)
+    @archived = false
+  end
+
+  def move_to_archive
+    return unless can_be_archived?
+
+    @archived = true
+  end
+
+  private
+
+  def can_be_archived?
+    age_in_years = Date.today.year - @publish_date.year
+    return true if age_in_years >= 10
+
+    false
+  end
```

Comment:
- [ ] This is not quite accurate, you're only comparing the difference of the year which is not always true to check if the game has 10+ years from publication. Instead, consider doing something like this: 🤓 🔧 

```rb
    def can_be_archived?
      today = Date.today

      # Create a date instance exactly 10 years ago
      ten_years_ago = Date.new(today.year - 10, today.month, today.day)

      # If the publish date is less (older) than ten_years_ago,
      # then return true and can be archived.
      @published_date < ten_years_ago
    end

```